docs/middleware/ics29-fee/fee-distribution.md

```diff
@@ -27,7 +27,7 @@ If a counterparty payee is not registered for the forward relayer on the destina
 A transaction must be submitted to the destination chain including a `CounterpartyPayee` address of an account on the source chain.
 The transaction must be signed by the `Relayer`.
 
-Note: If a module account address is used as the `CounterpartyPayee` it is recommended to [turn off invariant checks](https://github.com/cosmos/ibc-go/blob/71d7480c923f4227453e8a80f51be01ae7ee845e/testing/simapp/app.go#L659) for that module. 
+Note: If a module account address is used as the `CounterpartyPayee` it may be necessary to [turn off invariant checks](https://github.com/cosmos/ibc-go/blob/71d7480c923f4227453e8a80f51be01ae7ee845e/testing/simapp/app.go#L659) for that module, if the module has been set as a blocked address in the `BankKeeper`. This is because many modules use invariants to compare internal tracking of module account balances against the actual balance of the account stored in the `BankKeeper`. If a token transfer to the module account occurs without going through this module and updating the account balance of the module on the `BankKeeper`, then invariants may break and unknown behaviour could occur depending on the module implementation. Therefore, it is suggested that either invariant checks are turned off, or the module is removed from the blocked address list on the `BankKeeper` with the understanding that this may result in consequences for module balance accounting.
```

Comment:
So if I understand it correctly, there are two options here described that the use could choose from:

1. Either to turn off invariants.
2. Or to remove the module account from the list of blocked addresses.

If option 1 is chosen, is it still possible to use the module account as counterparty payee? Because in our code all the operations to transfer funds are using the bank keeper methods, so if the module is still in the list of blocked addresses, these operations will fail, right?

Reply:
correct, the code that is linked is to turn off invariant checks.

regarding your second question, yes it is possible to use the module account as the counterparty payee. each chain handles blocked addresses differently, so the operation may or may not fail depending on if the module is in the blocked address.

Reply:
>correct, the code that is linked is for the blocked addresses

So should we link instead to code that shows how to turn off invariants instead?

Reply:
I think I disagree with the recommended approach. If a module requires strict internal accounting (as shown by their usage of invariants), I don't think we can safely suggest to ignore that functionality. In my opinion, given the current state of SDK invariants/interaction with the bank keeper, we should say that module cannot be used as a counterparty payee

Reply:
one issue i see with that approach @colin-axner  might be for the community pool, feegrant module, budget module, or another custom module that a chain might want to use as a counterparty payee

Reply:
The lack of checks preventing payment to the community pool has led to multiple security releases (including one after the check preventing payment was removed). I think it is more sustainable to request changes to the SDK to support payment to modules without disrupting their internal tracking. This is a big ask, but the same question/problem has occurred for 3 years now and I'd prefer to prioritize security of UX

Reply:
Issues on the SDK, [#10811](https://github.com/cosmos/cosmos-sdk/issues/10811), [#11369](https://github.com/cosmos/cosmos-sdk/issues/11369), and [#11388](https://github.com/cosmos/cosmos-sdk/issues/11388) would all help make the requested functionality possible. Essentially you only want modules and their sub functionality blocked if it is a strict requirement for security. Currently some modules might be overly secure (too strict invariants) or may have sub-functionality that doesn't need to follow the same invariant checking (community pool)

Reply:
i see. one problem with waiting on the SDK to fix this though is that i think the usecase we have pinned down for the first deployments of fee middleware use module accounts as the payee, so it would be an issue if we somehow now technically block it. 

the other approach, where we don't technically block it but just STRONGLY recommend against and explain why module accounts are risky to use, still allows this usecase.

Reply:
Module accounts can be used, just module accounts which are blocked cannot be used

Reply:
@crodriguezvega is correct that turning off invariants is not sufficient to allow a blocked module to be used with ics29. The module account must be unblocked. Could we recommend that only module accounts which are not blocked can be used with ics29?